### ISSUE TITLE
Southern latitudes (and Western longitudes)

### ISSUE THREAD

**Issue by NigelTwo** [mobifu1/arduino_gps_clock#4]
Modifying this code to run "down under". The changes to create negative values for lat/long are trivial, and the sun position is then calculated and displayed correctly.
The moon() function seems to give bad results with negative latitude co-ordinates?
It is simple to fix the moon phase display (as the moon is viewed upside down in southern hemisphere ;-)).
Lastly, is the tidal "ham/cux_offset" the same as the "lunitidal interval"? If it is, then I can calculate it for my locale.

**Comment by longjohn119** [comment]
I had a little problem getting the program to compile, I kept getting an error related to the Time.h library. Apparently changes in the Arduino IDE borked the Time library so you need to update to Time v1.5 and change the #include line from Time.h to TimeLib.h ..... That fixed the compiling error except for a stupid one I made myself. Now I just need to figure out how to modify it for my area of the world (Midwest USA) and change the GPS serial code so it can use a Bluetooth module so all I'll have to do to hook it up in my little boat is run a cord to the USB hub in the console and it can grab the Bluetooth output off my GPS

**Comment by mobifu1** [comment]
hello boys, i am very busy in the next time to help you. but for a cross check, you can use my libs, stored here in this project.
thank you for your interest, regards Mobifu

**Comment by NigelTwo** [comment]
+longjohn119, It took a while before my local time was correct.
Adjust the timezone definitions at 48-49
But beware of the lines 321-326, they should probably read:
&tcr->offset/60
You will have to allow negative longitudes (West) to be negative in the GPS data. An E/W data check is necessary for parameter 4 and 6.
I would be interested to know how your values of sun and moon compare to:
www.suncalc.org
www.mooncalc.org
for your location.

**Comment by NigelTwo** [comment]
+longjohn119. My previous comment about the hard-coded tomezone parameters at lines 321-326 should have been:
  /* No need for this test as tcr points to the current time rule */
  //if (CE.locIsDST(cst)) { //ask for DST=Daylight saving time
  //  daylightsavingtime = 2; //true
  //}
  //else {
  //  daylightsavingtime = 1;//false
  //}
  daylightsavingtime = tcr->offset/60;

**Comment by longjohn119** [comment]
I think I have it worked out (famous last words)  but  with heat indexes in excess of 105-110 degrees last week it's just too hot in my garage to test it on on my boat's GPS  I can't get reception in my workshop which is in my basement without an external antenna and the 2 GPS's I have with support for external antennas are tied up  I found a case on Thingieverse that has just enough room in the top to put a Bluetooth module. No real hurry now because we just got hit here with another Global Warming Superstorm and the Mississippi River is going back above flood stage for the 2nd time in 2 months and barring any new rain it'll take a couple of weeks before conditions are good again ....

**Comment by longjohn119** [comment]
I finally got around to getting this working. Since the time zone is fixed/hard coded I decided there was no real sense in writing a routine to parse the NMEA sentence for "W" since it's always going to be West or -90.xxx .... So after the validity check ( line 352 " if ((lat > 0) && (lon > 0) && (lat < 90) && (lon < 180)) { " ) I just inserted the line " lon = -(lon); " .... yeah it's kinda cheating but it works 

However ...... I am thinking of trying to add a routine (If it can be done it's likely already out there somewhere) to automatically set the proper time zone based on position .... I say this because this would make a nice clock an an ocean going vessel. While I'm an inland fisherman I map lakes and rivers and have this plugged into a Raspberry Pi 3 running Openplotter and routed the $GNRMC sentence out that USB port 

I also saw in part of the code that was commented out how to adjust the sunrise/sunset times (I had to add about 8 minutes to sunrise and add 4 1/2 minutes to sunset) and while it's still a small sample size (6 days) so far it's been spot on with the National Weather Service .... and I also had to change $GPRMC to $GNRMC because that's what my GPS module (NEO M8N) outputs since it also receives GLONASS (and sometimes Galileo) satellites. So far the data on www.suncalc.org and www.mooncalc.org look to match reasonably close but like I noted it's still a small sample set
